### DISCUSSION TITLE
I want to know what algorithm is used by the join function?

### DISCUSSION THREAD

**Discussion by huntto** [CGAL/cgal#7857] [General]
https://doc.cgal.org/latest/Boolean_set_operations_2/Boolean_set_operations_2_2set_union_8cpp-example.html

**Comment by efifogel** [comment]
We use CGAL 2D arrangements to carry out (regularized) Boolean operations.
Each face of the arrangement is extended with a Boolean flag that indicates
whether it is included in the point set.
When computing the 'join' (or any other operation) we first construct an
arrangement for each polygon input.
Then, we use the CGAL::overlay() free function to overlay the arrangements
associated with the input polygons.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/
